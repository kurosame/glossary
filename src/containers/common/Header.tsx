import React from 'react'
import styled from 'styled-components'

export default class Header extends React.Component {
  public render() {
    return <Div>Header</Div>
  }
}

const Div = styled.div`
  color: white;
  background-color: red;
`
