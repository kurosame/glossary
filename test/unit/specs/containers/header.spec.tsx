import Header from '@/containers/common/Header'
import { shallow } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'

const wrapper = shallow(<Header />)
const wrapperRouter = shallow(
  <Router initialEntries={['/']}>
    <Header />
  </Router>
)

test('The state is initialized in the constructor', () => {
  expect(wrapper.state('tabValue')).toEqual(0)
})

test('Change the tabs will set the state.tabValue', () => {
  wrapper.find('.tabs').simulate('change', {}, 1)

  expect(wrapper.state('tabValue')).toEqual(1)
})

test('Match the snapshot', () => {
  expect(wrapperRouter.html()).toMatchSnapshot()
})
