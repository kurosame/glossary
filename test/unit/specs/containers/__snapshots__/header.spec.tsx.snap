// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot 1`] = `"<header class=\\"MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7\\"><div class=\\"MuiTabs-root-36\\" data-test=\\"category-tabs\\"><div><div style=\\"width:100px;height:100px;position:absolute;top:-10000px;overflow:scroll;-ms-overflow-style:scrollbar\\"></div></div><div class=\\"MuiTabs-flexContainer-37\\"><div class=\\"MuiTabs-scroller-38 MuiTabs-scrollable-40\\" style=\\"margin-bottom:0\\" role=\\"tablist\\"><div class=\\"MuiTabs-flexContainer-37\\"><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Summary</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/architecture\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Architecture</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/design\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Design</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/browser\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Browser</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/html\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">HTML</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/css\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">CSS</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/js\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">JS</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47 MuiTab-selected-50\\" role=\\"tab\\" aria-selected=\\"true\\" href=\\"/react\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">React</span></span></span><span class=\\"TabIndicator-root-67 TabIndicator-colorSecondary-69 MuiTabs-indicator-44\\"></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/vue\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Vue.js</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/swift\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Swift</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/python\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Python</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/ml\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">ML</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/apache\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Apache</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/server\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Server</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/network\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Network</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/package-manager\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Package Manager</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/module-bundler\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Module Bundler</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/compiler\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Compiler</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/aws\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">AWS</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/firebase\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Firebase</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/hardware\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Hardware</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/language\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Language</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/programming\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Programming</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/words\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Words</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47\\" role=\\"tab\\" aria-selected=\\"false\\" href=\\"/other\\"><span class=\\"MuiTab-wrapper-53\\"><span class=\\"MuiTab-labelContainer-54\\"><span class=\\"MuiTab-label-55\\">Other</span></span></span><span class=\\"MuiTouchRipple-root-60\\"></span></a></div></div></div></div></header>"`;
